ingest.py

```diff
@@ -65,7 +68,11 @@ def load_documents(source_dir: str) -> List[Document]:
         all_files.extend(
             glob.glob(os.path.join(source_dir, f"**/*{ext}"), recursive=True)
         )
-    return [load_single_document(file_path) for file_path in all_files]
+    pool = multiprocessing.Pool(os.environ.get('LOAD_DOCUMENTS_NUMBER_OF_THREADS', multiprocessing.cpu_count()))
```

Comment:
I don't think it is recommended to use every thread possible, since it can cause a lockup, I would at least subtract 1 from the cpu_count() probably more depending on the number someone has on their system.

Reply:
I kept the "- 1" from the example bellow.

I don't know if it's necessary to make a more complex code to calculate the number of threads, since the user can change the .env file, what do you think?

Reply:
Yes I agree